cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,177 @@
+                #  Specially for the VmDeployedEvent where the source_vm defers from the
+                # self.vm.name
+                if evt.destination_vm == self.vm.name and evt.event_type in self.tl_event:
+                    found_events.append(evt)
+                    break
+            elif not hasattr(evt, 'source_vm') or not hasattr(evt, 'source_host'):
+                logger.warn('Event %r does not have source_vm, source_host. Probably an issue', evt)
+            elif evt.source_vm == self.vm.name and evt.event_type in self.tl_event:
+                found_events.append(evt)
+                break
+
+        logger.info('found events on {tgt}: {evt}'.format(tgt=target, evt="\n".join([repr(e) for e
+                                                                                    in
+                                                                    found_events])))
+        return len(found_events)
+
+    def catch_in_timelines(self, soft_assert):
+        """Will execute self._check_timelines on multiple targets .
+
+
+        Returns:
+            Boolean
+
+        Raises:
+             :py:class:`TimedOutError` and message with category of failed event on its host.
         """
-        rest_vm = appliance.rest_api.collections.vms.get(name=vm)
-        event = dict(
-            status=fauxfactory.gen_alphanumeric(),
-            message=fauxfactory.gen_alphanumeric(),
-            event=fauxfactory.gen_alphanumeric(),
-        )
-        if from_detail:
-            responses = [rest_vm.action.add_lifecycle_event(**event)]
-        else:
-            responses = appliance.rest_api.collections.vms.action.add_lifecycle_event(
-                rest_vm, **event)
-        assert appliance.rest_api.response.status_code == 200
-        for response in responses:
-            assert response["success"] is True, "Could not add event"
-
-        # DB check
-        lifecycle_events = appliance.db.client["lifecycle_events"]
-        events_list = list(appliance.db.client.session.query(lifecycle_events).filter(
-            lifecycle_events.message == event["message"],
-            lifecycle_events.status == event["status"],
-            lifecycle_events.event == event["event"],
-        ))
-        assert events_list, "Could not find the lifecycle event in the database"
-
-
-@pytest.mark.manual
-def test_policy_events():
-    pass
-
-
-@pytest.mark.manual
-def test_timelines_ui():
-    pass
+        targets = (self.vm, self.vm.cluster, self.vm.host, self.vm.provider)
+        for target in targets:
+            try:
+                wait_for(self._check_timelines, [target], timeout='5m', fail_condition=0)
+            except TimedOutError:
+                soft_assert(False, '0 occurrence of {} found on the timeline of {}'.format(
+                    self.event, target))
+
+
+def test_timeline_events(new_vm, soft_assert):
+    events_list = ['create', 'stop', 'start', 'suspend', 'start']
```

Comment:
**REQUIRED**
This list should go to `@pytest.parametrize`

Reply:
@quarckster, it should be but I didn't manage to make it work. module fixtures fall down to function scope and etc.

Reply:
Darn scoping!